### DISCUSSION TITLE
Improved String whitespace utilities

### DISCUSSION THREAD

**Discussion by pbhogan** [pbhogan/sterile#18] [Ideas]
These are some whitespace utilities that I use all the time extracted from another codebase which could replace `trim_whitespace`. They could use better names perhaps. Unfortunately, `strip` has a [different meaning](https://ruby-doc.org/core-2.5.0/String.html#method-i-strip) in Ruby which I feel is actually a poor choice. In any event, these versions do a better job of dealing with whitespace, including unicode whitespace.

````
class String

  def trim_whitespace
    gsub(/^[\p{Z}\s]+/, "").gsub(/[\p{Z}\s]+$/, "").gsub(/[\p{Z}\s]+/, " ")
  end


  def trim_whitespace!
    gsub! /^[\p{Z}\s]+/, ""
    gsub! /[\p{Z}\s]+$/, ""
    gsub! /[\p{Z}\s]+/, " "
    self
  end


  def trim_html_whitespace
    # maybe a better way to implement this?
    decode_entities.trim_whitespace.encode_entities
  end


  def strip_whitespace
    gsub /[\p{Z}\s]+/, ""
  end


  def strip_whitespace!
    gsub! /[\p{Z}\s]+/, ""
  end

end
````

**Comment by gurgeous** [comment]
Neat, I like those. A few thoughts:

- I'd probably rename `trim_whitespace` to `squish_more` since it is very similar to [squish](https://apidock.com/rails/String/squish). One of my favorite helpers, btw.
- `squish_more` can just be `dup.squish_more!`
-  I might shy away from `trim_html_whitespace` unless you have a real specific use case... Is the goal to handle & nbsp;, & #32; etc? Enterprising users can probably call those three methods themselves without too much trouble. 
- Personally I'd find it a little confusing that `strip_whitespace` looks like [strip](https://ruby-doc.org/core-2.5.0/String.html#method-i-strip), but does something completely different. Maybe call it `remove_whitespace`? Oh, and `remove_whitespace` can call the bang version :)
- Actually, we could probably use a `strip_more` that works like [strip](https://ruby-doc.org/core-2.5.0/String.html#method-i-strip) (but uses `\p{Z}`).

(edited for typos and also the & ... was ironically getting unescaped)
